src/utils/OpenSearchMetricsQuery.groovy

```diff
@@ -9,7 +9,7 @@
 
 package utils
 
-import groovy.json.JsonSlurper
+import groovy.json.JsonSlurperClassic
```

Comment:
Had to use replace this due to serialization/de-serialization issue. [SO post](https://stackoverflow.com/questions/37864542/jenkins-pipeline-notserializableexception-groovy-json-internal-lazymap?source=post_page-----d1d22461b16a--------------------------------)

Reply:
This has been working with the current setup right Sayali?

Reply:
Sayali coming from here https://github.com/opensearch-project/opensearch-build-libraries/blob/main/vars/gradleCheckFlakyTestDetector.groovy#L55  `writeFile file: "${failedTest}.md", text: markdownTable` does this help to continue to use `groovy.json.JsonSlurper` ?
Because not sure when `JsonSlurperClassic` will be deprecated and how existing libraries behave with `JsonSlurperClassic`.

Reply:
Nope that is the exact step where it failed. Looks like writing to file would need serialization/de-serialization

Reply:
[queryData](https://github.com/opensearch-project/opensearch-build-libraries/pull/516/files#diff-d57b933f5803357c87cfda4fd0d3fe1425ff69adc6baa948c9375f5947194733R50) here appears to return the `LazyMap` which throws an error in jenkinsPipeline. I tried converting the response to string and back to JSON but looks like groovy automatically converts it to `LazyMap` due to its complexity. I think it is safe to go with `JsonSlurperClassic` for now as the queries can become more complex in the future.

Reply:
We can go with `JsonSlurperClassic` if there is no other way, only lib we need to test is `gradleCheckFlakyTestDetector.groovy`. But I assume it should be working fine.

Reply:
@gaiksaya upgrading this https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile to latest lib (that will be created once this PR is merged) should not break anything right ?

Reply:
Nope. We can keep the version same for above mentioned jenkinfile but even if we upgrade the breaking changes are related to build library. So gradle check should be fine.